payjoin/Cargo.toml

```diff
@@ -22,6 +22,8 @@ bitcoin = { version = "0.30.0", features = ["base64"] }
 bip21 = "0.3.1"
 log = { version = "0.4.14"}
 rand = { version = "0.8.4", optional = true }
+serde = { version = "1.0.107", features = ["derive"] }
```

Comment:
do we have better options than serde? 
I have some reservations in adding it to such a project for the two reasons:
1) a couple of months ago this thing happened https://www.bleepingcomputer.com/news/security/rust-devs-push-back-as-serde-project-ships-precompiled-binaries/
it was eventually reverted but it is a red flag
2) serde does a great job but it also makes compilation/building the project much slower

Reply:
We only need to parse and produce 2 lines of valid json without anything fancy like nesting. What do you suggest as an alternative, and why?

Reply:
we could consider https://github.com/rhysd/tinyjson 
1) its well documented, tested and the code is quite readable
2) its really tiny(we can also clone it into payjoin repo and strip out any code we dont need, we just parse string really)

disadvantages: last update was a few months ago, but that's not necessarily a red flag

I think optimal scenario is to clone tinyjson, clean it up so it has only functionality we need and maintain it ourselves. 

maybe we can also leave serde and put it behind a flag for now.

it really depends on how fast we want this pr to be merged


*I am intentionally not mentioning performance here because we are not dealing with large/nested json structures so that should be a problem

Reply:
regarding json-rust, its not maintained and the last update was 3 years ago, an alternative crates were developed(like rust-json)

Reply:
> I think optimal scenario is to clone tinyjson, clean it up so it has only functionality we need and maintain it ourselves.

This is definitely overkill to parse 4 lines of JSON that should only ever encode / decode strings with errorCode and message fields. We could honestly write an even simpler parser and forego a JSON dependency without a problem.

Reply:
We might miss some edge cases, but we could simplify ser/de do the following

```rs
fn serialize_error(error_code: &str, message: &str) -> String {
    format!(r#"{{"errorCode": "{}", "message": "{}"}}"#, error_code, message)
}
```

above is how we've already been serializing errors

```rs
fn deserialize_error(json: &str) -> Option<(&str, &str)> {
    let error_code_start = json.find("\"errorCode\": \"")? + "\"errorCode\": \"".len();
    let error_code_end = json.find("\", \"message\": \"")?;
    let message_start = error_code_end + "\", \"message\": \"".len();
    let message_end = json.rfind("\"}")?;

    let error_code = &json[error_code_start..error_code_end];
    let message = &json[message_start..message_end];

    Some((error_code, message))
}
```